.github/scripts/build-and-run-sanitizers.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# Copyright (c) 2019 Kevin R Croft <krcroft@gmail.com>
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# This simple script is used to build, test, and archive the log output from
+# various sanitizer-builds.
+#
+# Usage:
+#   ./build-and-run-sanitizers.sh COMPILER VERSION SANITIZER [SANITIZER [...]]"
+#
+# Examples:
+#   ./build-and-run-sanitizers.sh clang 8 msan usan
+#   ./build-and-run-sanitizers.sh gcc 9 asan uasan usan tsan
+#
+set -euo pipefail
+
+# Configurable compression settings:
+ZSTD_OPTS="--long=31 -19 -T0 -v"
+
+# Check the arguments
+if [[ "$#" -lt 3 ]]; then
+  echo "Usage: $0 COMPILER VERSION SANITIZER [SANITIZER [...]]"
+  exit 1
+fi
+
+# Accept the arguments
+compiler="${1}"
+compiler_version="${2}"
+shift; shift
+sanitizers=("${@}")
```

Comment:
That would make sense :frown:
```
$ a=(1 2 3)

$ echo ${a[1]}
echo ${a[1]}
2

$ b=${a[@]}

$ echo ${b[1]}
echo ${b[1]}

$ echo ${b[0]}
echo ${b[0]}
1 2 3
```

Reply:
Ah, I forgot to mention: `"$@"` expands to an array of quoted strings, which might seem surprising on first sight (I am sure shellcheck will insists on using quotes).

Reply:
Fixed in 257de204

Reply:
Yup, I've had to revert to the original code.
```
declare -A sanitizers
sanitizers="$@"
^-- SC2178: Variable was used as an array but is now assigned a string.
           ^-- SC2124: Assigning an array to a string! Assign as array, or use * instead of @ to 
```
This solves it: `sanitizers=("$@")`